kable-core/src/jvmMain/kotlin/com/juul/kable/btleplug/BtleplugAdvertisement.kt

```diff
@@ -0,0 +1,48 @@
+package com.juul.kable.btleplug
+
+import com.juul.kable.Identifier
+import com.juul.kable.ManufacturerData
+import com.juul.kable.PlatformAdvertisement
+import com.juul.kable.btleplug.ffi.PeripheralProperties
+import kotlinx.io.bytestring.ByteString
+import kotlin.uuid.Uuid
+
+internal data class BtleplugAdvertisement(
+    private val manufacturerDataMap: Map<UShort, ByteString>,
+    internal val serviceData: Map<Uuid, ByteString>,
+    override val name: String?,
+    override val peripheralName: String?,
+    override val identifier: Identifier,
+    override val rssi: Int,
+    override val txPower: Int?,
+    override val uuids: List<Uuid>,
+) : PlatformAdvertisement {
+
+    constructor(properties: PeripheralProperties) : this(
+        manufacturerDataMap = properties.manufacturerData.asSequence()
+            .associate { (key, value) -> key to ByteString(value) },
+        serviceData = properties.serviceData.asSequence()
+            .associate { (key, value) -> Uuid.parse(key) to ByteString(value) },
+        name = properties.localName,
+        peripheralName = properties.localName,
+        identifier = Identifier(properties.id),
+        rssi = properties.rssi?.toInt() ?: Int.MIN_VALUE,
+        txPower = properties.txPowerLevel?.toInt(),
+        uuids = properties.services.map(Uuid::parse),
+    )
+
+    override val isConnectable: Boolean?
+        get() = null // Doesn't exist in btleplug
+
+    // TODO: Figure out why Btleplug exposes this as a map.
```

Comment:
Re this comment, Web APIs also define a map. https://webbluetoothcg.github.io/web-bluetooth/#bluetoothmanufacturerdatamap

Not particularly useful but there you go. :-)

Reply:
> Instances of [BluetoothManufacturerDataMap](https://webbluetoothcg.github.io/web-bluetooth/#bluetoothmanufacturerdatamap) have a [[BackingMap]] slot because they are [maplike](https://webidl.spec.whatwg.org/#dfn-maplike),

Wow, you weren't kidding about not particularly useful.